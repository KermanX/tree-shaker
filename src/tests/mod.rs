use crate::{TreeShakeConfig, TreeShakeOptions};
use insta::{assert_snapshot, glob};
use oxc::{codegen::CodegenOptions, minifier::MinifierOptions};
use std::fs;

fn tree_shake(input: String) -> String {
  let do_minify = input.contains("@minify");
  let react_jsx = input.contains("@react-jsx");
  let result = crate::tree_shake(
    input,
    TreeShakeOptions {
      config: TreeShakeConfig::default().with_react_jsx(react_jsx),
      minify_options: do_minify.then(|| MinifierOptions::default()),
      codegen_options: CodegenOptions::default(),
    },
  );
  result.codegen_return.code
}

#[test]
fn test() {
  glob!("fixtures/**/*.js", |path| {
    let input = fs::read_to_string(path).unwrap();
    let mut settings = insta::Settings::clone_current();
    settings.set_prepend_module_to_snapshot(false);
    settings.bind(|| {
      assert_snapshot!(tree_shake(input));
    })
  });
}
