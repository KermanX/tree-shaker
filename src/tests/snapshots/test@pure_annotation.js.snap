---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/pure_annotation.js
---
/*#__PURE__*/ unknown1(1, other1());
/*#__PURE__*/ unknown2.g(
	1,
	other2(),
	/*#__PURE__*/ other3()
);
/*#__PURE__*/ unknown3?.g().h(1, other4());
export const a = unknown4.g?.(1, other5());
function simple() {
	effect();
	return a;
}
/*#__PURE__*/ simple();
/*#__PURE__*/ simple(other1());
export const b = /*#__PURE__*/ simple(other2());
export const c = /*#__PURE__*/ simple(
	0,
	/*#__PURE__*/ other2()
);
function nested1() {
	return /*#__PURE__*/ simple();
}
nested1(other1());
export const d = /*#__PURE__*/ nested1(other2());
function nested2() {
	unknown(1);
	return unknown(2);
}
/*#__PURE__*/ nested2(other1());
export const e = /*#__PURE__*/ nested2(other2());
function mutate(obj) {
	obj.a = b;
	return obj.f();
}
/*#__PURE__*/ mutate({ f: () => effect() });
export const f = /*#__PURE__*/ mutate({ f: effect });
class Class {
	constructor() {
		effect();
	}
}
/*#__PURE__*/ new Class(other1());
export const g = /*#__PURE__*/ new Class(other2());
